package/android/src/main/java/com/shopify/reactnative/skia/SkiaBaseView.java

```diff
@@ -17,42 +19,37 @@ public abstract class SkiaBaseView extends ReactViewGroup implements TextureView
 
     public SkiaBaseView(Context context) {
         super(context);
-        // TODO: Remove if we find another solution for first frame rendering
-        //setWillNotDraw(!shouldRenderFirstFrameAsBitmap());
         mTexture = new TextureView(context);
         mTexture.setSurfaceTextureListener(this);
         mTexture.setOpaque(false);
         addView(mTexture);
     }
 
-    /*@Override
-    TODO: Remove if we find another solution for first frame rendering
-    protected void onDraw(Canvas canvas) {
-        super.onDraw(canvas);
-
-        // If we haven't got a surface yet, let's ask the view to
-        // draw into a bitmap and then render the bitmap. This method
-        // is typically only called once - for the first frame, and
-        // then the surface will be available and all rendering will
-        // be done directly to the surface itself.
-        if (shouldRenderFirstFrameAsBitmap() && mSurface == null) {
-            int width = getWidth();
-            int height = getHeight();
-
-            if (width > 0 && height > 0) {
-                Bitmap bitmap = Bitmap.createBitmap(width, height, Bitmap.Config.ARGB_8888);
-                Bitmap result = (Bitmap) renderToBitmap(bitmap, width, height);
-
-                canvas.drawBitmap(
-                        result,
-                        new Rect(0, 0, width, height),
-                        new Rect(0, 0, width, height),
-                        null);
-
-                bitmap.recycle();
-            }
+    @Override
+    protected void finalize() throws Throwable {
+        // This API Level is >= 26, we created our own SurfaceTexture,
+        // onSurfaceTextureDestroyed won't be invoked by the texture view
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+            destroySurface();
         }
-    }*/
+    }
+
+    public void destroySurface() {
+        Log.i("SkiaBaseView", "destroySurface");
+        surfaceDestroyed();
+        mSurface.release();
+        mSurface = null;
+    }
+
+    public void createSurfaceTexture() {
+        // This API Level is >= 26, we created our own SurfaceTexture to have a faster time to first frame
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
```

Comment:
When suggesting this solution I missed the fact SurfaceTexture(bool) constructor only got added in API 26. This suggestion was based on what's happening in TextureView sources e.g. here: https://android.googlesource.com/platform/frameworks/base/+/master/core/java/android/view/TextureView.java#458 so if you are concerned about the coverage for older devices there are definitely similar ways of getting the surface created.